core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -661,4 +669,51 @@ public boolean equals(Object obj) {
       }
    }
 
+   public void testLockedStream() throws Throwable {
+      for (int i = 0; i < 10; i++) {
+         cache.put(i, "value" + i);
+      }
+
+      CyclicBarrier barrier = new CyclicBarrier(2);
+
+      int key = 4;
+
+      LockedStream<Object, Object> stream = cache.getAdvancedCache().lockedStream();
+      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
+         if (e.getKey().equals(key)) {
+            try {
+               barrier.await(10, TimeUnit.SECONDS);
+               assertEquals("value" + key, c.put(e.getKey(), String.valueOf(e.getValue() + "-other")));
+
+               barrier.await(10, TimeUnit.SECONDS);
+            } catch (InterruptedException | BrokenBarrierException | TimeoutException e1) {
+               throw new RuntimeException(e1);
+            }
+         }
+      }));
+
+      barrier.await(10, TimeUnit.SECONDS);
+
+      Future<Object> putFuture = fork(() -> cache.put(key, "value" + key + "-new"));
+      try {
+         putFuture.get(50, TimeUnit.MILLISECONDS);
```

Comment:
I really don't see what `expectException` would buy us here that the try catch doesn't already give us. The usage of `Exceptions` class in my mind is when you have a `Throwable` or something like that and you want to either unwrap it or assert the exact class instance of. The try catch idiom already guarantees that only a `TimeoutException` is caught so if any other is caught it will fail the test and if it isn't thrown the fail will complain as well.